gitpod-io/gitpod
# [Proof of Concept] Usage-based / pay-as-you-go with Stripe

## Description


A proof-of-concept using Stripe for usage-based / pay-as-you-go.

TODO:
- [x] Install Stripe in backend (server) and front-end (dashboard)
- [x] Add a `PaymentElement` (or simpler `CardElement`) to the dashboard for adding a payment method (requires a client secret)
- [x] When a payment method is added, create a new Strip Customer + pay-as-you-go Subscription
- [x] Send usage records to Stripe when a workspace stops

## Related Issue(s)


## How to test


1. Go to https://jx-stripe.preview.gitpod-dev.com/billing
2. Add a payment method
3. Verify in the Stripe dashboard that your customer was created and subscribed
4. Start a workspace, then stop the workspace manually
5. Verify in Stripe dashboard that a usage record was sent

## Release Notes

```release-note
NONE
```

## Documentation
